### PR TITLE
Support react native

### PR DESCRIPTION
**Pull request by davidliu**

### CODE REVIEW

**Review by davidzhao — COMMENTED** [review]
looks good! could you install eslint in VS Code and take a look at the lint failures?

File: src/api/SignalClient.ts [hunk 1]
```diff
@@ -20,9 +20,12 @@ import {
   UpdateSubscription, UpdateTrackSettings,
 } from '../proto/livekit_rtc';
 import { ConnectionError } from '../room/errors';
-import { getClientInfo, sleep } from '../room/utils';
 import Queue from './RequestQueue';
 
+if (isWeb()) {
+  import('webrtc-adapter');
```
Comment (davidliu) [inline src/api/SignalClient.ts:26]: This dynamic import is the main thing I'm unsure about. The webrtc-adapter library has a bunch of web specific code that crashes on react native, so needs to be omitted for RN. Didn't run into any issues on Chrome, but if there's a more proper way to do this, I'm all ears.

Comment (davidzhao) [reply src/api/SignalClient.ts:26]: I'm not sure if there's a better way.. will test this for safari and FF to ensure things are still working.

**Review by davidzhao — APPROVED** [review]
confirming it works on FF/Safari too.

**Pull request merged**